File: Modules/ServerTypes/TaskFlow/TaskDetailForm.ts
import { initFormType, IntegerEditor, LookupEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";
import { PersianDateEditor } from "../../Common/PersianDateEditor";

export interface TaskDetailForm {
    Title: StringEditor;
    Description: StringEditor;
    TaskStateId: LookupEditor;
    DueDate: PersianDateEditor;
    Duration: IntegerEditor;
    TagIds: ServiceLookupEditor;
}

export class TaskDetailForm extends PrefixedContext {
    static readonly formKey = 'TaskFlow.TaskDetail';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!TaskDetailForm.init) {
            TaskDetailForm.init = true;

            initFormType(TaskDetailForm, [
                'Title', StringEditor,
                'Description', StringEditor,
                'TaskStateId', LookupEditor,
                'DueDate', PersianDateEditor,
                'Duration', IntegerEditor,
                'TagIds', ServiceLookupEditor
            ]);
        }
    }
}

File: Modules/TaskFlow/Task/TaskDetailForm.cs
namespace TaskFlowSerenity.TaskFlow.Forms;

[FormScript("TaskFlow.TaskDetail")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskDetailForm
{
    public string Title { get; set; }

    public string Description { get; set; }

    public Guid TaskStateId { get; set; }

    public DateTime? DueDate { get; set; }

    public int Duration { get; set; }

    public List<Guid> TagIds { get; set; }
}